dataactvalidator/config/sqlrules/a32_appropriations.sql

```diff
@@ -0,0 +1,18 @@
+SELECT approp.row_number,
+	approp.allocation_transfer_agency,
+	approp.agency_identifier,
+	approp.beginning_period_of_availa,
+	approp.ending_period_of_availabil,
+	approp.availability_type_code,
+	approp.main_account_code,
+	approp.sub_account_code
+FROM appropriation AS approp
+WHERE approp.submission_id = {0}
+	AND EXISTS (
+		SELECT a.row_number,
+			a.tas
+		FROM appropriation AS a
+		WHERE a.row_number <> approp.row_number
+			AND a.tas IS NOT DISTINCT FROM approp.tas
+			AND a.submission_id = {0}
+	);
```

Comment:
We had a lot of back and forth about the best way to surface errors for duplicate rows, since we don't have a way to know which row is the "right" one. I'm fine with reporting all rows involved, as long as we use that approach consistently for all rules related to dupes.

Agreed with @cmc333333 that a different alias for readability would be nifty.